src/darwin/Framework/CHIPTests/MTRXPCListenerSampleTests.m

```diff
@@ -81,9 +81,43 @@ @implementation MTRXPCListenerSample
 
 - (instancetype)init
 {
+    // Classes that are used by MTRDevice responses.  In particular, needs to
+    // include NSError.
+    //
+    // TODO: should this include NSOrderedSet, NSSet, NSDate, NSNull, even
+    // though we don't use them?  Should we have public API for some sort for
+    // this set?
+    __auto_type * allowedClasses = [NSSet setWithArray:@[
+        [NSString class], [NSNumber class], [NSData class], [NSArray class], [NSDictionary class], [NSError class]
+    ]];
     if ([super init]) {
         _serviceInterface = [NSXPCInterface interfaceWithProtocol:@protocol(MTRDeviceControllerServerProtocol)];
         _clientInterface = [NSXPCInterface interfaceWithProtocol:@protocol(MTRDeviceControllerClientProtocol)];
+        [_serviceInterface setClasses:allowedClasses
```

Comment:
Can we factor out a function each of our client and server protocols that creates the `NSXPCInterface` and configures these parameters? It seems strange have to have this duplicated into the test.

Reply:
Well, I wrote the bit above before reading this....

We can certainly do that.  Would we just have them as free functions, or would they hang off of something else?